.github/workflows/tag-stable-commit.yml

```diff
@@ -0,0 +1,175 @@
+            const minCommitLength= 7;      // typical short git commit hash length
+
+            // Add minimum length requirement to validate output variables          
+            if (!codeName || codeName.length < minCodeNameLength) {
+              core.setFailed(`Invalid codeName: "${codeName}"`);
+              return;
+            }        
+            if (!version || version.length < minVersionLength) {
+              core.setFailed(`Invalid version: "${version}"`);
+              return;
+            }
+            if (!commit || commit.length < minCommitLength) {
+              core.setFailed(`Invalid commit: "${commit}"`);
+              return;
+            }
+
+            core.setOutput('codeName', codeName);
+            core.setOutput('version', version);
+            core.setOutput('commit', commit);
+            core.info(`codeName=${codeName}`);
+            core.info(`version=${version}`);
+            core.info(`commit=${commit}`);
+
+      # Verify commit exists in the repo and print commit message first line
+      - name: Verify commit exists
+        id: verify
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const target = '${{ steps.snap.outputs.commit }}'.slice(0, 7);  // 7-char prefix
+            core.info(`Looking for a commit starting with "${target}" ...`);
+            const commits = await github.paginate(
+              github.rest.repos.listCommits,
+              {
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                per_page: 100
+              }
+            );
+            const hit = commits.find(c => c.sha.startsWith(target));
+            if (hit) {
+              core.info(`Found commit: ${hit.sha} - ${hit.html_url}`);
+              // Print first line of the commit message
+              const firstLine = hit.commit.message.split('\n')[0];
+              core.info(`Commit message first line: "${firstLine}"`);
+              core.setOutput('full_sha', hit.sha); // Output full SHA
+            } else {
+              core.setFailed(`No commit starting with "${target}" found in the repository`);
+            }
+
+      # Create or update tag pointing to the verified commit
+      # If the tag doesn't exist, create it
+      # If the tag exists, but it points to a different commit, update it (move it forward)
+      # If it already exists but points to the correct commit, do nothing     
+      - name: Create or update stable tag for verified commit
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const commitSha = '${{ steps.verify.outputs.full_sha }}';
+            const codeName = '${{ steps.snap.outputs.codeName }}';
+            const version = '${{ steps.snap.outputs.version }}';
+            const stableTag = `v${version}+${codeName}`;
```

Comment:
So from what I've read above, if you use `v${Version}+${codeName}` as the tag, an example tag could look like: 
-> `v19.2.0+squid` right ?

This has a slight problem that the tag will change everytime the underlying ceph point release changes. for example: in future the tag could become:
-> `v19.2.2+squid`. 

Now since the tag is changing based on ceph-release, one of us would have to manually go and pull a new version on the RTD dashboard based on the new tag.

I propose we do not use the full semantic version but just the constant bit (something like `v19+squid`) as it will stay constant throughout the squid release life. And every time this workflow re-tags a commit, the documentation will update automatically.